### PR TITLE
Track the address range of inlined function instances

### PR DESCRIPTION
**Pull request by jebradbury39**
The value here is that by keeping track of the addresses, it allows for exact unrolling of the inlined function during analysis (for example, determining which inlined function instance called a given address).

Also updated some of the dependencies while I was at it.

### CODE REVIEW

File: parser/src/function.rs [hunk 1]
```diff
@@ -349,7 +350,25 @@ impl<'input> InlinedFunction<'input> {
         Function::from_offset(hash, self.abstract_origin)
     }
 
-    /// The size of the inlined function.
+    /// The address of the inlined function instance.
+    #[inline]
+    pub fn address(&self) -> Option<u64> {
```
Comment (philipc) [inline parser/src/function.rs:355]: I don't think it makes sense to associate a single address with an `InlinedFunction`, since it doesn't have an entry point like a `Function` does.

File: parser/src/function.rs [hunk 2]
```diff
@@ -349,7 +350,25 @@ impl<'input> InlinedFunction<'input> {
...
+        self.address.get()
+    }
+
+    /// The address ranges of the function.
+    pub fn range(&self) -> Option<Range> {
+        if let (Some(address), Some(size)) = (self.address(), self.size()) {
+            Some(Range {
+                begin: address,
+                end: address + size,
```
Comment (philipc) [inline parser/src/function.rs:364]: This is incorrect for instances that have multiple ranges. We should do the same as we already do for `Function`: store `Vec<Range>` in the struct have a `ranges` method that returns `&[Range]`.

File: main/Cargo.toml [hunk 3]
```diff
@@ -12,15 +12,15 @@ edition = "2018"
...
 marksman_escape = "0.1"
 parser = { package = "ddbug_parser", version = "0.3.0", path = "../parser" }
-hyper = { version = "1.2.0", features = ["server", "http1"] }
-hyper-util = { version = "0.1.3", features = ["tokio"] }
-http-body-util = "0.1.0"
-tokio = { version = "1.37.0", features = ["net", "rt"] }
+hyper = { version = "1.4.1", features = ["server", "http1"] }
+hyper-util = { version = "0.1.7", features = ["tokio"] }
+http-body-util = "0.1.2"
+tokio = { version = "1.39.2", features = ["net", "rt"] }
```
Comment (philipc) [inline main/Cargo.toml:23]: Dependency updates should be a separate PR.

**Review by philipc — CHANGES_REQUESTED** [review]
Thanks for the PR.

File: parser/src/function.rs (hunk 2 above)
Comment (jebradbury39) [reply parser/src/function.rs]: Done

File: parser/src/function.rs (hunk 1 above)
Comment (jebradbury39) [reply parser/src/function.rs]: Done (added ranges field instead).

**Review by philipc — APPROVED** [review]
Thanks!
